File: IT/2/Input-Output/Esercizi/Python/esercizio24.py
# file esercizio24.py

from __future__ import print_function
if hasattr(__builtins__, 'raw_input'): input = raw_input

filename = input("Apri il file: ")

try:
	doc = open(filename, "r")
	s = input("Parola da cercare: ")
	counter = 0

	for line in doc:
		counter += line.lower().count(s.lower())

	if counter != 0:
		print("La parola \"{}\" compare {} volte nel file.".format(s, counter))
	else:
		print("Parola \"{}\" non trovata!".format(s))

	doc.close()

except IOError as e:
	print(e)

File: IT/2/Input-Output/Esercizi/Java/Esercizio24.java
/* file Esercizio24.java */

import java.io.Console;
import java.io.File;
import java.io.FileReader;
import java.io.BufferedReader;
import java.io.FileNotFoundException;
import java.io.IOException;

public class Esercizio24 {
	public static void main(String [] args) {
		Console console = System.console();
		String filename = console.readLine("Apri il file: ");

		try {
			File doc = new File(filename);

			FileReader reader = new FileReader(doc);
			BufferedReader buffer = new BufferedReader(reader);

			String s = console.readLine("Parola da cercare: ");

			int counter = 0;

			while (buffer.ready()) {
				String line = buffer.readLine().toLowerCase();
				int pos = 0;
				while ((pos = line.indexOf(s.toLowerCase(), pos)) != -1) {
					counter++;
					pos++;
				}
			}

			if (counter != 0) {
				System.out.println("La parola \"" + s + "\" compare " + counter + " volte nel file.");
			}
			else {
				System.out.println("Parola \"" + s + "\" non trovata!");
			}

			buffer.close();
		}
		catch (FileNotFoundException e) {
			System.out.println(e);
		}
		catch (IOException e) {
			System.out.println(e);
		}
	}
}

